### ISSUE TITLE
Containers do not always start fully

### ISSUE THREAD

**Issue by achekulaev** [docksal/docksal#187]
In some cases containers start but the following 2 lines:
```
...
_set_cli_uid && \
_vhost_proxy_connect
```
Are not executed. This leads to issues with file permissions or site access.

This is how it looks when it fails:

<img width="805" alt="no-start" src="https://cloud.githubusercontent.com/assets/402007/25352402/e6ed9698-28f9-11e7-8e75-eb9498ad2beb.png">

This is how it should be:

<img width="533" alt="yes-start" src="https://cloud.githubusercontent.com/assets/402007/25352440/07d11cfe-28fa-11e7-8771-f98babf020c8.png">

**Comment by achekulaev** [comment]
So conditions do work. The thing is that permissions change happens only once, when uid is not set. Basically once after container was created and never again.

I don't know if we should consider doing it **every time**. As people might be confused too. Here is what I got in chat today:

> When installing gulp and node modules during init, sometimes docksal returns error about 
> permissions (ownership) of some folders and that prevents init to be completed. In order to install things properly i must manually set ownership using `fin exec sudo chown -R docker /home/docker`. 
>After that, everything runs normally. This happens on mac.

**Comment by achekulaev** [comment]
Most like it is related to #225 . Closing in favor of that one
